source/extensions/filters/http/stateful_session/stateful_session.h

```diff
@@ -66,11 +87,50 @@ class StatefulSession : public Http::PassThroughFilter,
   // Http::StreamEncoderFilter
   Http::FilterHeadersStatus encodeHeaders(Http::ResponseHeaderMap& headers, bool) override;
 
+  // Http::StreamFilterBase
+  Http::LocalErrorStatus onLocalReply(const Http::StreamFilterBase::LocalReplyData&) override {
+    // For strict mode, if an override was attempted and a local reply was sent (e.g., 503),
+    // consider this a failed-closed selection.
+    if (override_attempted_ && effective_config_ != nullptr && effective_config_->isStrict() &&
+        !accounted_) {
+      markFailedClosed();
+      accounted_ = true;
+    }
+    return Http::LocalErrorStatus::Continue;
+  }
```

Comment:
I think this is a little weird because any other local responses may be handled here incorrectly. You may should count the failed closed requests at the `encodeHeaders` by checking whether there is a valid upstream info and response flag is UH or not?